src/main/java/seedu/planner/model/ModulePlanner.java

```diff
@@ -451,6 +437,152 @@ private void sortAvailableModules(List<Module> modulesAvailable, UserProfile use
         return allModules;
     }
 
+    private Optional<ModuleDescription> getModuleDescription(String code) {
+        return MajorDescription.getModuleCode(userProfile.getMajor(), code);
+    }
+
+    /**
+     * Count the number of general education modules
+     */
+    private void countGeneralEducation() {
+        int count = 0;
+        for (Module m : getAllModulesTaken()) {
+            if (m.toString().startsWith("GER") || m.toString().startsWith("GEQ")
+                || m.toString().startsWith("GET") || m.toString().startsWith("GEH")
+                || m.toString().startsWith("GES")) {
+                count += m.getCreditCount();
+            }
+        }
+        statusMap.put(DegreeRequirement.UNIVERSITY_LEVEL_REQUIREMENTS, new int[] {count});
+    }
+
+    /**
+     * Count the number of foundation modules.
+     */
+    private void countFoundation() {
+        int count = 0;
```

Comment:
Can extract out lines 463 to 470 into a method since the same piece of code is being used for the many of the other `count` methods as well

Reply:
Done